pkg/plugin/plugin.go

```diff
@@ -19,10 +18,6 @@ type Plugin struct{}
 
 var _ api.Plugin = &Plugin{}
 
-func (p *Plugin) ValidateExternal(oc *v1.OpenShiftCluster) []error {
```

Comment:
The reason for removing this is to combine Create and Update validations into one? Also create validation will happen only on converted internal model and not the external request from the customer?

Reply:
No, this was based on our architecture discussion where we agreed to only validate the internal types.  Step 7 of the create process and step 6 of the upgrade process in the document.

Reply:
My response may be ambiguous :smile: 

> reason for removing this is to combine Create and Update validations into one

No, this was based on our architecture discussion where we agreed to only validate the internal types. 

> create validation will happen only on converted internal model and not the external request from the customer?

Correct, as discussed, MSFT will do a basic validation on external during create and update - Step 4 and 3 respectively in the doc.  MSFT and RH will do internal validations - step 7 and 6 respectively

Reply:
Step 4 was for MSFT to do validation on things like core quota count, SPN, or maybe DNS FQDN in the future. But validation for others parameters in PUT request body still needs to be done which should be in the plugin.

If this combined how will we know which what is create vs update?

Reply:
`ValidateInternal(new, old *ContainerService) []error`

if there is an old container service it is an update